xclim/indices/_hydrology.py

```diff
@@ -345,15 +345,15 @@ def high_flow_frequency(
     Returns
     -------
     xr.DataArray
-        Calculated mean of high flow days per water year
+        Number of high flow days.
 
     References
     ----------
     :cite:cts:`addor2018,Clausen2000`
     """
-    median_flow = q.median(dim="time")
-    threshold = threshold_factor * median_flow
     sel = select_time(q, **indexer)
+    median_flow = sel.median(dim="time")
+    threshold = threshold_factor * median_flow
```

Comment:
In that case, you could also remove the `indexer` here and simply create the indicator by inheriting from `ResamplingIndexingIndicator`.

But the result would be the same at the indicator level, so your choice.

Reply:
I removed indexing alltogether. I'm not sure anymore it even makes sense in those cases.